api/openapi.yaml

```diff
@@ -1213,6 +1213,487 @@ paths:
+        "404":
+          $ref: "#/components/responses/NotFoundProblemResponse"
+        "412":
+          $ref: "#/components/responses/PreconditionFailedProblemResponse"
+        default:
+          $ref: "#/components/responses/UnexpectedProblemResponse"
+
+  # Notifications - Rules
+
+  /api/v1/notifications/rules:
+    get:
+      operationId: listNotificationRules
+      summary: List notification rules
+      description: |
+        List all notification rules.
+      tags:
+        - Notifications (Experimental)
+      parameters:
+        # FIXME: add filtering based on type, features and subjects
+        - $ref: "#/components/parameters/queryLimit"
+        - $ref: "#/components/parameters/queryOffset"
+        - name: orderBy
+          in: query
+          required: false
+          description: Order by field
+          schema:
+            type: string
+            enum:
+              - id
+              - createdAt
+              - updatedAt
+            default: id
+        - name: includeArchived
+          in: query
+          required: false
+          description: Include archived notification channels.
+          schema:
+            type: boolean
+            default: false
+      responses:
+        "200":
+          description: List of notification rules.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/NotificationRulesListResponse"
+        "400":
+          $ref: "#/components/responses/BadRequestProblemResponse"
+        "401":
+          $ref: "#/components/responses/UnauthorizedProblemResponse"
+        default:
+          $ref: "#/components/responses/UnexpectedProblemResponse"
+    post:
+      operationId: createNotificationRule
+      summary: Create a notification rule
+      description: |
+        Create a new notification rule.
+      tags:
+        - Notifications (Experimental)
+      requestBody:
+        description: The notification channel to rule.
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/NotificationRuleCreateInputs"
+      responses:
+        "201":
+          description: Notification channel created.
+          content:
+            application/json:
+              schema:
+                oneOf:
+                  - $ref: "#/components/schemas/NotificationRuleBalanceThreshold"
+          headers:
+            ETag:
+              $ref: "#/components/headers/ETag"
+        "400":
+          $ref: "#/components/responses/BadRequestProblemResponse"
+        "401":
+          $ref: "#/components/responses/UnauthorizedProblemResponse"
+        "409":
+          $ref: "#/components/responses/ConflictProblemResponse"
+        "501":
+          $ref: "#/components/responses/NotImplementedProblemResponse"
+        default:
+          $ref: "#/components/responses/UnexpectedProblemResponse"
+
+  /api/v1/notifications/rules/{ruleId}:
```

Comment:
So I do not think we can change the parameter name used for referencing as it nneds to be unique, but we definitely can change it on the API level.

Like this:
```
     ruleId:
-      name: ruleId
+      name: id
       description: A unique ULID identifier for a notification rule.
       in: path
       required: true
```

I'll update the spec, as I like this approach more.

Reply:
I have testing this approach and decided not to proceed with renaming resource identifiers in path to `id` because:
* it would not be aligned with rest of the spec, so I prefer not to have mixed styles
* this works only for endpoints with single `id` parameter